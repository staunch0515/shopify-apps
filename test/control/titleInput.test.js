import { AppProvider } from '@shopify/polaris';

import React from "react";
import { mount } from 'enzyme';

import TitleInput from "../../components/control/titleInput.js"

let wrapper;

describe('TitleInput component', () => {

    beforeAll(() => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(), // deprecated
                removeListener: jest.fn(), // deprecated
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            })),
        });
    });

    beforeEach(() => {

        const setValue = (value, valid) => {
            if (value && value.length > 2) {
                expect(valid).toBe(true);
            } else {
                expect(valid).toBe(false);
            }
        }

        wrapper = mount(
            <AppProvider>
                <TitleInput label="test" text="test001" setValue={setValue} autoFocus={true} />
            </AppProvider>
        );

        expect(wrapper).toMatchSnapshot();

    });

    it('initialling', () => {
        // initialling
        const label = wrapper.find("label");
        expect(label.text()).toBe('test');

        const input = wrapper.find("input");
        expect(input.instance().value).toBe('test001');
    });

    it('check value changed', () => {
        const input = wrapper.find("input");
        // receive value
        input.getDOMNode().value = 'test003';
        input.simulate('change', input);
    });

    it('check value changed', () => {
        const input = wrapper.find("input");
        // receive value
        input.getDOMNode().value = 't';
        input.simulate('change', input);
    });

    it('check value changed', () => {
        const input = wrapper.find("input");
        // receive value
        input.getDOMNode().value = 'ZDFeiell';
        input.simulate('change', input);
    });
});