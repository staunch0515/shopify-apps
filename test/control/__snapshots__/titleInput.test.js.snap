// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleInput component check value changed 1`] = `
<div
  style={
    Object {
      "--p-frame-offset": "0px",
      "--top-bar-background": "#00848e",
      "--top-bar-background-lighter": "#1d9ba4",
      "--top-bar-color": "#f9fafb",
    }
  }
>
  Array [
    "",
    <div
      className=""
    >
      <div
        className="Polaris-Labelled__LabelWrapper"
      >
        <div
          className="Polaris-Label"
        >
          <label
            className="Polaris-Label__Text"
            htmlFor="PolarisTextField1"
            id="PolarisTextField1Label"
          >
            test
          </label>
        </div>
      </div>
      <div
        className="Polaris-Connected"
      >
        <div
          className="Polaris-Connected__Item Polaris-Connected__Item--primary"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="Polaris-TextField Polaris-TextField--hasValue"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <input
              aria-describedby="PolarisTextField1HelpText"
              aria-invalid={false}
              aria-labelledby="PolarisTextField1Label"
              aria-multiline={false}
              autoFocus={true}
              className="Polaris-TextField__Input Polaris-TextField__Input--alignLeft"
              disabled={false}
              id="PolarisTextField1"
              maxLength={20}
              minLength={2}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="入力してください"
              style={null}
              value="test001"
            />
            <div
              className="Polaris-TextField__Backdrop"
            />
          </div>
        </div>
      </div>
      <div
        className="Polaris-Labelled__HelpText"
        id="PolarisTextField1HelpText"
      >
        全半角文字title・2～20文字
      </div>
    </div>,
  ]
</div>
`;

exports[`TitleInput component check value changed 2`] = `
<div
  style={
    Object {
      "--p-frame-offset": "0px",
      "--top-bar-background": "#00848e",
      "--top-bar-background-lighter": "#1d9ba4",
      "--top-bar-color": "#f9fafb",
    }
  }
>
  Array [
    "",
    <div
      className=""
    >
      <div
        className="Polaris-Labelled__LabelWrapper"
      >
        <div
          className="Polaris-Label"
        >
          <label
            className="Polaris-Label__Text"
            htmlFor="PolarisTextField1"
            id="PolarisTextField1Label"
          >
            test
          </label>
        </div>
      </div>
      <div
        className="Polaris-Connected"
      >
        <div
          className="Polaris-Connected__Item Polaris-Connected__Item--primary"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="Polaris-TextField Polaris-TextField--hasValue"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <input
              aria-describedby="PolarisTextField1HelpText"
              aria-invalid={false}
              aria-labelledby="PolarisTextField1Label"
              aria-multiline={false}
              autoFocus={true}
              className="Polaris-TextField__Input Polaris-TextField__Input--alignLeft"
              disabled={false}
              id="PolarisTextField1"
              maxLength={20}
              minLength={2}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="入力してください"
              style={null}
              value="test001"
            />
            <div
              className="Polaris-TextField__Backdrop"
            />
          </div>
        </div>
      </div>
      <div
        className="Polaris-Labelled__HelpText"
        id="PolarisTextField1HelpText"
      >
        全半角文字title・2～20文字
      </div>
    </div>,
  ]
</div>
`;

exports[`TitleInput component check value changed 3`] = `
<div
  style={
    Object {
      "--p-frame-offset": "0px",
      "--top-bar-background": "#00848e",
      "--top-bar-background-lighter": "#1d9ba4",
      "--top-bar-color": "#f9fafb",
    }
  }
>
  Array [
    "",
    <div
      className=""
    >
      <div
        className="Polaris-Labelled__LabelWrapper"
      >
        <div
          className="Polaris-Label"
        >
          <label
            className="Polaris-Label__Text"
            htmlFor="PolarisTextField1"
            id="PolarisTextField1Label"
          >
            test
          </label>
        </div>
      </div>
      <div
        className="Polaris-Connected"
      >
        <div
          className="Polaris-Connected__Item Polaris-Connected__Item--primary"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="Polaris-TextField Polaris-TextField--hasValue"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <input
              aria-describedby="PolarisTextField1HelpText"
              aria-invalid={false}
              aria-labelledby="PolarisTextField1Label"
              aria-multiline={false}
              autoFocus={true}
              className="Polaris-TextField__Input Polaris-TextField__Input--alignLeft"
              disabled={false}
              id="PolarisTextField1"
              maxLength={20}
              minLength={2}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="入力してください"
              style={null}
              value="test001"
            />
            <div
              className="Polaris-TextField__Backdrop"
            />
          </div>
        </div>
      </div>
      <div
        className="Polaris-Labelled__HelpText"
        id="PolarisTextField1HelpText"
      >
        全半角文字title・2～20文字
      </div>
    </div>,
  ]
</div>
`;

exports[`TitleInput component initialling 1`] = `
<div
  style={
    Object {
      "--p-frame-offset": "0px",
      "--top-bar-background": "#00848e",
      "--top-bar-background-lighter": "#1d9ba4",
      "--top-bar-color": "#f9fafb",
    }
  }
>
  Array [
    "",
    <div
      className=""
    >
      <div
        className="Polaris-Labelled__LabelWrapper"
      >
        <div
          className="Polaris-Label"
        >
          <label
            className="Polaris-Label__Text"
            htmlFor="PolarisTextField1"
            id="PolarisTextField1Label"
          >
            test
          </label>
        </div>
      </div>
      <div
        className="Polaris-Connected"
      >
        <div
          className="Polaris-Connected__Item Polaris-Connected__Item--primary"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="Polaris-TextField Polaris-TextField--hasValue"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            <input
              aria-describedby="PolarisTextField1HelpText"
              aria-invalid={false}
              aria-labelledby="PolarisTextField1Label"
              aria-multiline={false}
              autoFocus={true}
              className="Polaris-TextField__Input Polaris-TextField__Input--alignLeft"
              disabled={false}
              id="PolarisTextField1"
              maxLength={20}
              minLength={2}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="入力してください"
              style={null}
              value="test001"
            />
            <div
              className="Polaris-TextField__Backdrop"
            />
          </div>
        </div>
      </div>
      <div
        className="Polaris-Labelled__HelpText"
        id="PolarisTextField1HelpText"
      >
        全半角文字title・2～20文字
      </div>
    </div>,
  ]
</div>
`;
